Dockerfile

```
@@ -147,10 +148,14 @@ RUN rm -rf /usr/bin/python && ln /usr/bin/python3.7 /usr/bin/python
 
 RUN curl --silent https://bootstrap.pypa.io/get-pip.py | python
 
-# Set up ubuntu user
+# Set up ubuntu user and home directory with access to users in the root group (0)
 
-RUN groupadd --gid 1000 ubuntu \
-  && useradd --uid 1000 --gid ubuntu --shell /bin/bash --create-home ubuntu
+ENV HOME=/home/ubuntu
+RUN groupadd --gid 1000 ubuntu && \
+  useradd --uid 1000 --gid ubuntu --groups 0 --shell /bin/bash --no-create-home ubuntu && \
```

Create the ubuntu user and add it to both the ubuntu  group and the root group (0)

---

Does this have the affect that Renovate essentially then runs with root privileges, even though it's configured to run as a non-root user?

---

No. The root group unlike the user with the same name does not have any special permissions.
>Adding the following to your Dockerfile sets the directory and file permissions to allow users in the root group to access them in the built image:
>[...]
>Because the container user is always a member of the root group, the container user can read and write these files. The root group does not have any special permissions (unlike the root user) so there are no security concerns with this arrangement.

---

```suggestion
  useradd --uid 1000 --gid ubuntu --groups 0 --shell /bin/bash --create-home ubuntu && \
```

Shouldn't we leave creating home to `useradd` and only change perms after?

---

> Shouldn't we leave creating home to `useradd` and only change perms after?

I thought that it was more explicit by creating the directory ourselves.
Otherwise it is basically the same except for one difference that I can think of; `useradd` also adds the home directory to the user listing in `/etc/passwd`. This might actually be necessary for some programs, but hopefully not in our case. If it were to be necessary for Renovate to operate, we should do this for OpenShift too. [This guide explains how it could be done](https://docs.openshift.com/container-platform/3.6/creating_images/guidelines.html) in the "Support Arbitrary User IDs"-section.

How about `useradd --uid 1000 --gid ubuntu --groups 0 --shell /bin/bash --home-dir ${HOME} --create-home ubuntu && \` to make it slightly more explicit than relying on convention for the home directory creation? @ViceIce